### PR TITLE
Added show me questions for adi2

### PR DESCRIPTION
**Pull request by edwardwilson**
# Description and relevant Jira numbers
MES-4821 show me questions were missed from the schema

## Pull Request checklist

- [ ] [WIP] tag removed from PR title
- [ ] PR has an understandable description

## Git feature branch checklist

- [ ] branch name comply with our branching strategy
- [ ] git branch contains relevant JIRA ticket number
- [ ] branch rebased against the latest develop

## Sign off process checklist

- [ ] Code has been tested manually
- [ ] Tested by QA
- [ ] PO's approval
- [ ] Reviewers selected in Github
- [ ] PR link added to JIRA

### CODE REVIEW

File: mes-test-schema/categories/ADI2/partial.d.ts [hunk 1]
```diff
@@ -23,27 +15,22 @@ export interface PartialTestResultADI2Schema {
...
+    complete?: boolean;
+    /**
+     * Whether the candidate has failed the eyesight test
+     */
+    seriousFault?: boolean;
+    /**
+     * Comments recorded against a fault
+     */
+    faultComments?: string;
+  };
```
Comment (rossfauldsbjss) [inline mes-test-schema/categories/ADI2/partial.d.ts]: Why have we lost EyesightTest type?

File: mes-test-schema/categories/ADI2/partial.d.ts [hunk 2]
```diff
@@ -69,72 +56,156 @@ export interface TestRequirements {
...
+    /**
+     * Comments recorded against a fault
+     */
+    observationFaultComments?: string;
+    /**
+     * Indicator for a manoeuvre being performed during the test
+     */
+    selected?: boolean;
+  };
 }
```
Comment (rossfauldsbjss) [inline mes-test-schema/categories/ADI2/partial.d.ts]: Why have we lost the manouevresOutcome type.

File: mes-test-schema/categories/ADI2/partial.d.ts [hunk 3]
```diff
@@ -69,72 +56,156 @@ export interface TestRequirements {
...
+    /**
+     * Description of the question that was asked
+     */
+    description?: string;
+    /**
+     * Outcome of the question that was asked
+     */
+    outcome?: "P" | "DF" | "S" | "D";
+  }[];
 }
```
Comment (rossfauldsbjss) [inline mes-test-schema/categories/ADI2/partial.d.ts]: Why have we lost QuestionResultType?

**Comment by edwardwilson** [comment]
The types that have been moved to be implicit are due to the fact they are not exported in the ADI union type. Before the types, which are common types and come from the common schema, were declared as explicit types for ADI2 which is misleading - they are not types of ADI2 and we are not exposing them.

To use these types required a reference to common in, therefore the types in ADI2 partial redundant. These changes clean up and simplify the schema and make clear where the types come from and how to reference them in the mes-mobile-app project - in my opinion.

**Comment by AndyPerryKainos** [comment]
> The types that have been moved to be implicit are due to the fact they are not exported in the ADI union type. Before the types, which are common types and come from the common schema, were declared as explicit types for ADI2 which is misleading - they are not types of ADI2 and we are not exposing them.
> 
> To use these types required a reference to common in, therefore the types in ADI2 partial redundant. These changes clean up and simplify the schema and make clear where the types come from and how to reference them in the mes-mobile-app project - in my opinion.

(holds-hands-up) I intentionally didn't re-export some of the common types when I made the first pass on this because I incorrectly assumed that they would be re-exported implicitly and not output as literal types. IMO we should re-export these missing types for consistency.

**Comment by edwardwilson** [comment]
@AndyPerryKainos So you think we should redeclare then, even though they aren't export in the types to be consistent; though it's not necessary?

**Comment by AndyPerryKainos** [comment]
@edwardwilson I think that would be my preference, yes. Worth Szabi's confirmation as well though.

**Comment by rossfauldsbjss** [comment]
I'm concerned that things are being considered on a category by category basis - rather than taking  existing practices into consideration.  Supportability and maintainability should be important aspect and having a consistent approach and recognisable types is part of that.  I will not be approving without these types being added back in - or for the changes to be applied and tested across all categories (which is inadvisable at this stage)

File: mes-test-schema/category-definitions/ADI2/partial.json [hunk 4]
```diff
@@ -5,25 +5,40 @@
     "eyesightTest": {
       "$ref": "../common/index.json#/definitions/eyesightTest"
     },
+    "faultComments": {
```
Comment (szabikr) [inline mes-test-schema/category-definitions/ADI2/partial.json]: why do you have to declare these here if you're using the common definition down below anyway?

**Pull request merged**